### PR TITLE
Fix numeric category error with Postgres

### PR DESCRIPTION
**Pull request by matthieu-foucault**
#2981 ensured that numeric IDs that are passed in as strings by the front-end are converted to numbers. This needs to be done for numeric categories as well, as postgres will not do an implicit cast and fail to run the query otherwise.

AFAIK, there is no open issue for this bug, there are a few comments though, referenced by @flamber [here](https://github.com/metabase/metabase/issues/7891#issuecomment-497898971): 
 - https://github.com/metabase/metabase/issues/5632#issuecomment-489165061 @pelletencate
 - https://github.com/metabase/metabase/issues/2854#issuecomment-409295946 @thomashs

### CODE REVIEW

**Comment by matthieu-foucault** [comment]
PS: you can see that the test is failing in f3e545c here: https://github.com/bcgov/cas-ggircs-metabase/commits/fix/category-number-cast

File: src/metabase/query_processor/middleware/parameters/mbql.clj [hunk 1]
```diff
@@ -17,9 +17,9 @@
   convert the parameters to integers or floating-point numbers."
   [param-type param-value, field-id :- su/IntGreaterThanZero]
   (cond
-    ;; for `id` type params look up the base-type of the Field and see if it's a number or not. If it *is* a number
-    ;; then recursively call this function and parse the param value as a number as appropriate.
-    (and (= param-type :id)
+    ;; for `id` or `category` type params look up the base-type of the Field and see if it's a number or not.
+    ;; If it *is* a number then recursively call this function and parse the param value as a number as appropriate.
+    (and (or (= param-type :id) (= param-type :category))
```
Comment (camsaul) [inline src/metabase/query_processor/middleware/parameters/mbql.clj]: In Clojure a common idiom for this situation is to do 

```clj
(#{:id :category} param-type)
```

`#{:id :category}` = a set literal 
passing an arg to a set in clojure returns the arg if it is in the set or `nil` if it is not.

**Review by camsaul — APPROVED** [review]
works for me

**Comment by camsaul** [comment]
Running the full test suite in #10112, once that's green this is ready to merge

File: src/metabase/query_processor/middleware/parameters/mbql.clj (hunk 1 above)
Comment (matthieu-foucault) [reply src/metabase/query_processor/middleware/parameters/mbql.clj]: Thanks for the tip! I am new to Clojure so I don't know the idioms yet. I just updated the PR with that change.

**Pull request merged**